src/adapters/github.js

```diff
@@ -149,10 +153,98 @@ class GitHub extends PjaxAdapter {
   _getTree(path, opts, cb) {
     this._get(`/git/trees/${path}`, opts, (err, res) => {
       if (err) cb(err)
-      else cb(null, res.tree)
+      else {
+        if (!opts.repo.pull) cb(null, res.tree)
+        else {
+          this._getPatch(opts, (patchErr, patchRes) => {
+            const diffExists = patchRes && Object.keys(patchRes).length > 0
+            if (patchErr || !diffExists) cb(null, res.tree)
+            else {
+              // Filter tree to only include files and directories that are included in the patch
+              const filteredTree = res.tree
```

Comment:
@bigredwill sounds good. In that case, I suppose when PR mode is used, the lazy mode is not used. I think that's even better, avoid one source of bug and it isn't necessary to lazy load PR changes as lazy load only makes sense for huge repo, not a single change set.